### ISSUE TITLE
Fix DT kselftest failures on mt8186-steelix

### ISSUE THREAD

**Issue by nfraprado** [kernelci/kernelci-project#400]
[Current](https://staging.kernelci.org:9000/viewer?search=parent%3D6679d5d3af336ef8c0144b36) DT kselftest failures on linux-next:
- [ ] `/soc/dpi@1400a000`, `/soc/i2c@1100f000/dp-bridge@5c`: The logs shows errors `it6505 3-005c: can not get extcon device!` and `platform 1400a000.dpi: deferred probe pending: mediatek-dpi: Failed to get bridge`. Applying [this old series](https://lore.kernel.org/all/20230331091145.737305-1-treapking@chromium.org/#t) (actually those commits + fixup from [this branch](https://gitlab.collabora.com/google/chromeos-kernel/-/tree/for-kernelci?ref_type=heads), since they no longer apply) fixed the failure;
- [ ] `/soc/i2c@11009000/trackpad@2c`: Second source component like the trackpad on juniper. Depends on series https://lore.kernel.org/all/20231128084236.157152-1-wenst@chromium.org/ being merged to be fixed.
- [ ] `/sound`: On the logs there's `platform sound: deferred probe pending: mt8186_mt6366: I2S3: codec dai not found`. From DT, I2S3's codecs are the it6505 dp-bridge and rt1019 speaker-codec, but even with both probing the issue persists. Note that [UCM](https://github.com/alsa-project/alsa-ucm-conf/pull/368) hasn't been merged upstream yet, so working audio is blocked on that, but this isn't required to get the driver probing.

**Comment by musamaanjum** [comment]
On recent [next](https://kcidb.kernelci.org/d/test/test?orgId=1&var-datasource=default&var-build_architecture=arm64&var-build_config_name=defconfig&var-id=maestro:672213183e51423f55a4c8ca), I've found new failures in addition to above mentioned failures:
- [ ] /soc/aal@1400c000
- [ ] /soc/ccorr@1400b000
- [ ] /soc/color@14009000
- [ ] /soc/dsi-phy@11cc0000
- [ ] /soc/dsi@14013000
- [ ] /soc/gamma@1400d000
- [ ] /soc/i2c@11007000/anx7625@58
- [ ] /soc/i2c@11007000/anx7625@58/aux-bus/panel
- [ ] /soc/ovl@14005000
- [ ] /soc/ovl@14006000
- [ ] /soc/rdma@14007000
- [ ] /soc/rdma@1401f000
- [ ] /soc/svs@1100bc00
